terraform/api_gateway.tf.json.template.py

```diff
@@ -254,16 +214,153 @@ def for_domain(cls, domain):
                                     }
                                 }
                                 for name, action, ip_set_term in [
-                                    ('BlockedIPs', 'block', config.blocked_v4_ips_term),
-                                    ('AllowedIPs', 'allow', config.allowed_v4_ips_term)
+                                    ('blocked_ips', 'block', config.blocked_v4_ips_term),
+                                    ('allowed_ips', 'allow', config.allowed_v4_ips_term)
                                 ]
                             ],
+                            {
+                                'name': config.waf_rate_rule_manifests.name,
+                                'statement': {
+                                    'rate_based_statement': {
+                                        'limit': config.waf_rate_rule_manifests.limit,
```

Comment:
Note that it is easy to accidentally 

```suggestion
                                        'limit': config.waf_rate_rule.limit,
```

which would be very difficult to detect and diagnose. The singleton comprehension would help with this, too.